### DISCUSSION TITLE
Does OTIO have an adaptor for .stl files (subtitles)?

### DISCUSSION THREAD

**Discussion by utsab** [AcademySoftwareFoundation/OpenTimelineIO#973] [Ideas]
I was looking for potential sample data to use in a OTIO tutorial.  I'm looking at the Netflix open source dataset that @jminor presented at the last meeting, specifically this [example](http://download.opencontent.netflix.com/?prefix=Meridian/).  The files are quite large, so I decided to start with the [subtitles data](http://download.opencontent.netflix.com/?prefix=Meridian/subtitles/STL/).  I'm noticing that the subtitles are stored in .stl files.  

Upon reading the [documentation](https://opentimelineio.readthedocs.io/en/latest/tutorials/adapters.html) on available adaptors, it seems that there is no adaptor for taking in .stl files.  I'm wondering if it would be worthwhile for me to create such an adaptor.  Do you think that there is a need in the OTIO community to read in .stl files?

**Comment by KarthikRIyer** [comment]
I was actually working on this sometime ago. There are a few issues with this. You can see my PR #805

**Reply by KarthikRIyer** [reply]
https://github.com/PixarAnimationStudios/OpenTimelineIO/issues/62#issuecomment-808361249

The actual problem is mentioned here

**Reply by meshula** [reply]
(just added a new note to help move that conversation along)

**Reply by utsab** [reply]
Thank you @KarthikRIyer for posting the link to your SRT adaptor. It looks like your adaptor can read in an .srt file.  However, in the Netflix dataset, the subtitles are provided as .stl files which I don't think is the same format as an srt file.   Thus, it seems that I would not be able to use your adaptor to read in an stl file.  

Upon closer inspection, I realized that the [stl files](http://download.opencontent.netflix.com/?prefix=Meridian/subtitles/STL/) are shared in a binary format.  According to this [website](https://forum.videohelp.com/threads/288632-How-to-open-and-edit-stl-file), I would need to use a propriety (not free) software to view and edit a binary stl file. 

Question: Should I continue exploring the possibility of importing an stl file into OTIO, or is it not worth the effort?   I'm not sure how common stl files are in the industry.  It seems that Netflix uses them.  Is this the common practice for subtitles to use binary stl files?

**Reply by meshula** [reply]
I would propose discussing this question with the community during the next TSC meeting. @reinecke ~ hereby flagging a topic for TSC :)

**Reply by reinecke** [reply]
It has been added to the TSC agenda.

**Reply by KarthikRIyer** [reply]
In case we need to implement an adapter and are not able to use ffmpeg,
STL subtitles specification: https://tech.ebu.ch/docs/tech/tech3264.pdf

**Reply by jminor** [reply]
Here is some background information on the differences between subtitles, closed captions, and audio description, as well as the use cases for each one: https://kb.iu.edu/d/awuf

**Comment by utsab** [comment]
In the last OTIO zoom meeting, we discussed the possibility of investigating how ffmpeg parses stl files.  It turns out that yes ffmpeg does in fact parse stl files but unfortunately it is a different kind of "stl" file (Spruce STL).  This is different from the .stl file used in the Netflix dataset which is the "EBU STL format".   If we do decide to proceed with creating an adaptor for the "EBU STL format" we may need to look at the [EBU STL specification](https://tech.ebu.ch/docs/tech/tech3264.pdf) which @KarthikRIyer posted earlier (unless someone has another idea).
